components/tyk-bootstrap/templates/_helpers.tpl

```diff
@@ -0,0 +1,90 @@
+{{- /*
+Expand the name of the chart.
+*/}}
+{{- define "tyk-bootstrap.name" -}}
+{{- default .Chart.Name .Values.bootstrap.nameOverride | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+
+{{- /*
+ok
+Create a default fully qualified app name.
+We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+If release name contains chart name it will be used as a full name.
+*/}}
+{{- define "tyk-bootstrap.fullname" -}}
+{{- if .Values.bootstrap.fullnameOverride -}}
+{{- .Values.bootstrap.fullnameOverride | trunc 63 | trimSuffix "-" -}}
+{{- else -}}
+{{- $name := default .Chart.Name .Values.bootstrap.nameOverride -}}
+{{- if contains $name .Release.Name -}}
+{{- .Release.Name | trunc 63 | trimSuffix "-" -}}
+{{- else -}}
+{{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+{{- end -}}
+{{- end -}}
+
+{{- /*
+Create chart name and version as used by the chart label.
+*/}}
+{{- define "tyk-bootstrap.chart" -}}
+{{- printf "%s-%s" .Chart.Name .Chart.Version | replace "+" "_" | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+
+{{- define "tyk-bootstrap.gwproto" -}}
+{{- if .Values.bootstrap.gwTls -}}
+https
+{{- else -}}
+http
+{{- end -}}
+{{- end -}}
+
+{{- define "tyk-bootstrap.dash_proto" -}}
+{{- if .Values.bootstrap.tls -}}
+https
+{{- else -}}
+http
+{{- end -}}
+{{- end -}}
+
+{{- define "tyk-bootstrap.dash_url" -}}
+{{ include "tyk-bootstrap.dash_proto" . }}://dashboard-svc-{{.Release.Name}}-tyk-dashboard.{{ .Release.Namespace }}.svc.cluster.local:{{ .Values.bootstrap.dashboard.service.port }}
+{{- end -}}
+
+{{- define "tyk-bootstrap.gateway_url" -}}
```

Comment:
Should we get url of gateway and dash from values file instead of assuming it?

Reply:
I think I will modify the logic so that if provided, usese the given url of gw/dash but if not it defaults to this pattern.